c/dynmat.c

```diff
@@ -567,16 +579,25 @@ static void get_dm(double dm[3][3][2], const long num_patom,
 }
 
 static double get_dielectric_part(const double q_cart[3],
-                                  const double dielectric[3][3]) {
+                                  const double dielectric[3][3],
+                                  const double diel_ext,
+                                  const double vacuum_size, const long dim){
     long i, j;
-    double sum;
+    double sum, dim_factor, q_norm;
 
     sum = 0;
     for (i = 0; i < 3; i++) {
         for (j = 0; j < 3; j++) {
             sum += q_cart[i] * dielectric[i][j] * q_cart[j];
         }
     }
+
+    if (dim == 2){
+        q_norm =  2*PI*sqrt(q_cart[0]*q_cart[0] + q_cart[1]*q_cart[1] + q_cart[2]*q_cart[2]);
+        dim_factor = (2./vacuum_size)*q_norm*diel_ext;
+        sum += dim_factor + sum/4.;
+    }
```

Comment:
Is the new implementation for the calculation, and are the others all interface?

Reply:
Yes, basically this is the core change for calculation. It would be great to have your input for the interface. I also just realized that implemented calculation is not correct - I have to verify and check carefully against QE implementation (probably one also needs an extra parameter). 

I implemented this like 1.5 years ago, and since it relies on PBC I thought to make the PR now. It would be good to have your feedback, happy to change and adapt. Overall it is a quite simple addition, once the units are on spot.